server/controllers/user.ts

```diff
@@ -1,40 +1,48 @@
 import express from 'express'
 import { User } from '../models/user'
+import {Avatar} from '../models/avatar'
 
 
 async function getUserData(req: express.Request, res: express.Response) {
     try {
-        const { email } = req.oidc.user;
+        const  {email,sub} = req.oidc.user
+        console.log('request opened to retrieve information from: ',email)
         const user = await User.findOne({ email: email })
         if (user) {
-            res.status(201).json({ "statusCode": 200, "message": "Retrieving information of the user", user: req.oidc.user });
+            const resUser = await User.findOne({ email: email }).populate('avatar');
+            res.status(201).json({ "statusCode": 201, "message": "Retrieving information of the user", user: resUser });
         } else {
+            const username = email.substr(0,email.indexOf('@'));
+            const avatar = await Avatar.findOne({contentType: 'monkey.png'});
```

Comment:
Can you just find the avatar id directly rather than getting the entire document?

Reply:
Changed to {id}